### ISSUE TITLE
"SCHEME_DOMAIN_PORT is not defined" failures due to HTTP 500 all over websockets/

### ISSUE THREAD

**Issue by foolip** [w3c/web-platform-tests#10457]
As found in https://github.com/w3c/web-platform-tests/issues/1695#issuecomment-381071483, lots of tsets under websockets/ are failing. For example:
https://wpt.fyi/websockets/binary

In Chrome devtools I see "Failed to load resource: the server responded with a status of 500 (Internal Server Error)" so something pretty odd is happening here.

**Comment by foolip** [comment]
@jugglinmike, not sure if this will come up in your investigation, but a widespread failure like this is pretty easy to justify the time spending to fix...

**Comment by zcorpan** [comment]
I'm assuming that a 500 can happen if there's no port defined for `wss`. `constants.js` contains:

```
var PORT_SSL = "{{ports[wss][0]}}";
```

**Comment by zcorpan** [comment]
[config.default.json](https://github.com/w3c/web-platform-tests/blob/master/config.default.json) (read by tools/wptserve/wptserve/config.py) has:
```
 "ports":{"http":[8000, "auto"],
          "https":[8443],
          "ws":["auto"],
          "wss":["auto"]},
```
I don't know what isn't working, though...

**Comment by gsnedders** [comment]
Labelling this as roadmap and not urgent on the assumption that this has been broken a while.

**Comment by gsnedders** [comment]
```
 0:37.96 WARNING list index out of range
Traceback (most recent call last):
  File "/Users/gsnedders/Documents/other-projects/wpt/web-platform-tests/tools/wptserve/wptserve/server.py", line 256, in handle_one_request
    handler(request, response)
  File "/Users/gsnedders/Documents/other-projects/wpt/web-platform-tests/tools/wptserve/wptserve/handlers.py", line 151, in __call__
    response = wrap_pipeline(path, request, response)
  File "/Users/gsnedders/Documents/other-projects/wpt/web-platform-tests/tools/wptserve/wptserve/handlers.py", line 117, in wrap_pipeline
    response = pipeline(request, response)
  File "/Users/gsnedders/Documents/other-projects/wpt/web-platform-tests/tools/wptserve/wptserve/pipes.py", line 34, in __call__
    response = func(request, response, *args)
  File "/Users/gsnedders/Documents/other-projects/wpt/web-platform-tests/tools/wptserve/wptserve/pipes.py", line 127, in inner
    return f(request, response, *arg_values)
  File "/Users/gsnedders/Documents/other-projects/wpt/web-platform-tests/tools/wptserve/wptserve/pipes.py", line 363, in sub
    new_content = template(request, content, escape_type=escape_type)
  File "/Users/gsnedders/Documents/other-projects/wpt/web-platform-tests/tools/wptserve/wptserve/pipes.py", line 440, in template
    new_content = template_regexp.sub(config_replacement, content)
  File "/Users/gsnedders/Documents/other-projects/wpt/web-platform-tests/tools/wptserve/wptserve/pipes.py", line 425, in config_replacement
    value = value[item[1]]
IndexError: list index out of range

 0:37.99 WARNING list index out of range
Traceback (most recent call last):
  File "/Users/gsnedders/Documents/other-projects/wpt/web-platform-tests/tools/wptserve/wptserve/server.py", line 256, in handle_one_request
    handler(request, response)
  File "/Users/gsnedders/Documents/other-projects/wpt/web-platform-tests/tools/wptserve/wptserve/handlers.py", line 151, in __call__
    response = wrap_pipeline(path, request, response)
  File "/Users/gsnedders/Documents/other-projects/wpt/web-platform-tests/tools/wptserve/wptserve/handlers.py", line 117, in wrap_pipeline
    response = pipeline(request, response)
  File "/Users/gsnedders/Documents/other-projects/wpt/web-platform-tests/tools/wptserve/wptserve/pipes.py", line 34, in __call__
    response = func(request, response, *args)
  File "/Users/gsnedders/Documents/other-projects/wpt/web-platform-tests/tools/wptserve/wptserve/pipes.py", line 127, in inner
    return f(request, response, *arg_values)
  File "/Users/gsnedders/Documents/other-projects/wpt/web-platform-tests/tools/wptserve/wptserve/pipes.py", line 363, in sub
    new_content = template(request, content, escape_type=escape_type)
  File "/Users/gsnedders/Documents/other-projects/wpt/web-platform-tests/tools/wptserve/wptserve/pipes.py", line 440, in template
    new_content = template_regexp.sub(config_replacement, content)
  File "/Users/gsnedders/Documents/other-projects/wpt/web-platform-tests/tools/wptserve/wptserve/pipes.py", line 425, in config_replacement
    value = value[item[1]]
IndexError: list index out of range
```

**Comment by gsnedders** [comment]
More breakage from bfef1f20a419d24633e48d24c14e6a7503e1d48c (#10231) it would seem.

**Comment by gsnedders** [comment]
So not broken for so long, and my fault…

**Comment by gsnedders** [comment]
@zcorpan `constants.js` seems to be fine.

**Comment by gsnedders** [comment]
OK, so it is `/websockets/constants.js?pipe=sub` that is failing with `500` (which makes sense in the case of `001.html`). This is wptrunner specific, seemingly, so presumably our override config is broken.

**Comment by gsnedders** [comment]
The bug here is:

```
        config.ports = {
            "http": [8000, 8001],
            "https": [8443],
            "ws": [8888]
        }
```
in `environment.py`.

**Comment by jgraham** [comment]
I need to fix this to unbreak a gecko sync, so there will be a patch soon.

**Comment by alijuma** [comment]
Pinging for an update since this is marked Priority:urgent.

**Comment by foolip** [comment]
Assigning to @jgraham as the urgency is on part of Gecko sync.

**Comment by gsnedders** [comment]
I reviewed a patch last week by @jgraham. Should we not fix it upstream soon, even if the Gecko sync doesn't land?

**Comment by gsnedders** [comment]
fixed in #10557 (which sadly didn't link here)
